ddl/ddl.go

```diff
@@ -207,6 +210,17 @@ func NewDDL(store kv.Storage, infoHandle *infoschema.Handle, hook Callback, leas
 	return newDDL(store, infoHandle, hook, lease)
 }
 
+// TODO: Move this to the function of newDDL.
+func (d *ddl) setWorker(ctx goctx.Context, cli *clientv3.Client) {
+	d.worker = &worker{
+		ddlID:      d.uuid,
+		etcdClient: cli,
+	}
+	ctx, d.worker.cancelFunc = goctx.WithCancel(ctx)
+	d.wait.Add(1)
+	go d.campaignOwners(ctx)
```

Comment:
it's confusing.
We can run `d.compaignOwners` directly instead of using 'go'.
New two childCtx and go the two campaignLoops inside it.